src/toil/jobStores/abstractJobStore.py

```diff
@@ -54,6 +58,47 @@ def __init__(self, message):
         super(JobStoreCreationException, self).__init__(message)
 
 
+@memoize
+def getJobStoreClasses():
+    """
+    Compiles as list of the classes of all jobStores whose dependencies
+    are installed.
+
+    Note that job store names must be manually added.
+    """
+    jobStoreClassNames = (
+        "toil.jobStores.azureJobStore.AzureJobStore",
+        "toil.jobStores.fileJobStore.FileJobStore",
+        "toil.jobStores.aws.jobStore.AWSJobStore"
+    )
+
+    jobStoreClasses = []
+    for className in jobStoreClassNames:
+        moduleName, className = className.rsplit('.', 1)
+        from importlib import import_module
+        try:
+            module = import_module(moduleName)
+        except ImportError:
+            logger.info("Unable to import '%s'" % moduleName)
+        else:
+            jobStoreClasses.append(getattr(module, className))
+    return jobStoreClasses
+
+
+def findJobStoreForUrl(url):
+    """
+    This method returns the class of the job store which supports exporting to and from the given
+    URL.
+
+    :param urlparse.ParseResult url:
+    :return: The class of the job store which supports this URL.
```

Comment:
"The concrete AbstractJobStore subclass that supports the given URL"

Add :rtype: annotation